drivers/ipm/ipm_nrfx_ipc.c

```diff
@@ -133,7 +135,8 @@ static void vipm_dispatcher(uint32_t event_mask, void *p_context)
 		event_mask &= ~BIT(event_idx);
 		if (nrfx_vipm_data.callback[event_idx] != NULL) {
 			nrfx_vipm_data.callback[event_idx]
-				(nrfx_vipm_data.callback_ctx[event_idx],
+				(nrfx_vipm_data.ipm_device,
```

Comment:
The `ipm_device` field is currently not initialized. It is actually not used at all (looks like it's some remnant of an earlier version of the driver). You'll need to initialize it in `vipm_nrf_init()` to use it this way (and perhaps rename it to `dev`, to avoid confusion).

Reply:
indeed, done

Reply:
I'm sorry, now I see I was wrong in my previous comment. This field should be changed to an array of `struct device *` and its entries should be initialized in `vipm_nrf_##_idx##_register_callback` (like `callback[_idx]` and `callback_ctx[_idx]` are), not in `nrfx_vipm_data`.